src/gc/gc.cpp

```diff
@@ -28986,6 +29249,12 @@ void gc_heap::mark_through_cards_for_segments (card_fn fn, BOOL relocating)
                             card, end_card,
                             foundp, start_address,
                             limit, total_cards_cleared);
+#ifdef FEATURE_CARD_MARKING_STEALING
+                        if (passed_end_card_p && !foundp)
+                        {
+                            foundp = find_next_chunk(card_mark_enumerator, seg, n_card_set, start_address, limit, card, end_card, card_word_end);
+                        }
+#endif // FEATURE_CARD_MARKING_STEALING
```

Comment:
seems like this should just be part of the card_transition function 'cause it is part of the card transition...

Reply:
Yeah, you're right. I was reluctant to pass even more parameters to card_transition, but it's probably better that way...

Reply:
Done.